src/ert/_c_wrappers/enkf/queue_config.py

```diff
@@ -15,182 +15,41 @@
-    _has_job_script = ResPrototype("bool queue_config_has_job_script( queue_config )")
-    _get_job_script = ResPrototype("char* queue_config_get_job_script(queue_config)")
-    _max_submit = ResPrototype("int queue_config_get_max_submit(queue_config)")
-    _queue_system = ResPrototype("char* queue_config_get_queue_system(queue_config)")
-    _queue_driver = ResPrototype(
-        "driver_ref queue_config_get_queue_driver(queue_config, char*)"
-    )
-    _get_num_cpu = ResPrototype("int queue_config_get_num_cpu(queue_config)")
-
-    _lsf_queue_opt = ResPrototype("char* queue_config_lsf_queue_name()", bind=False)
-    _lsf_server_opt = ResPrototype("char* queue_config_lsf_server()", bind=False)
-    _lsf_resource_opt = ResPrototype("char* queue_config_lsf_resource()", bind=False)
-    _lsf_driver_opt = ResPrototype("char* queue_config_lsf_driver_name()", bind=False)
-
-    def __init__(
-        self,
-        config_content: Optional[ConfigContent] = None,
-        config_dict=None,
-    ):
-        configs = sum(1 for x in [config_content, config_dict] if x is not None)
-
-        if configs > 1:
-            raise ValueError(
-                "Attempting to create QueueConfig object with multiple config objects"
-            )
-
-        if configs == 0:
-            raise ValueError(
-                "Attempting to create QueueConfig object with no config objects"
-            )
-
-        c_ptr = None
-        if config_content is not None:
-            c_ptr = self._alloc_content(config_content)
-        if config_dict is not None:
-            job_script_path = shutil.which("job_dispatch.py") or "job_dispatch.py"
-            c_ptr = self._alloc_full(
-                config_dict.get(ConfigKeys.JOB_SCRIPT, job_script_path),
-                config_dict.get(ConfigKeys.USER_MODE, True),
-                config_dict.get(ConfigKeys.MAX_SUBMIT, 2),
-                config_dict.get(ConfigKeys.NUM_CPU, 0),
-                config_dict.get(ConfigKeys.QUEUE_SYSTEM, LOCAL_DRIVER),
-            )
-        if not c_ptr:
-            raise ValueError("Unable to create QueueConfig instance")
-
-        super().__init__(c_ptr)
-
-        # Need to create
-        if config_dict is not None:
-            queue_options = config_dict.get(ConfigKeys.QUEUE_OPTION, [])
-            for option in queue_options:
-                self.driver.set_option(
-                    option[ConfigKeys.NAME], option[ConfigKeys.VALUE]
-                )
+from dataclasses import dataclass, field
+from typing import Dict, List, Tuple, Union
+
+from ert._c_wrappers.job_queue import Driver, JobQueue, QueueDriverEnum
+
+
+@dataclass
+class QueueConfig:
+    job_script: str = shutil.which("job_dispatch.py") or "job_dispatch.py"
+    max_submit: int = 2
+    num_cpu: int = 0
+    queue_system: QueueDriverEnum = QueueDriverEnum.NULL_DRIVER
+    queue_options: Dict[
+        QueueDriverEnum, List[Union[Tuple[str, str, str], Tuple[str, str]]]
+    ] = field(default_factory=dict)
```

Comment:
It means the default value is empty dict.

Reply:
That is because that would set the parameter to one dict for all instances of QueueConfig, we need one dict per instance. See for instance https://stackoverflow.com/questions/43255481/python-default-list-in-function

Reply:
ah - because it's executed during reading of the function definition. I see! thanks : )